yt_dlp/plugins.py

```diff
@@ -11,20 +11,55 @@
 import sys
 import traceback
 import zipimport
+import functools
 from pathlib import Path
 from zipfile import ZipFile
 
+from ._globals import (
+    plugin_dirs,
+    all_plugins_loaded,
+    plugin_specs,
+    plugins_enabled,
+    Indirect,
+)
+
 from .utils import (
-    Config,
     get_executable_path,
     get_system_config_dirs,
     get_user_config_dirs,
+    merge_dicts,
     orderedSet,
     write_string,
+    YoutubeDLError,
 )
 
 PACKAGE_NAME = 'yt_dlp_plugins'
 COMPAT_PACKAGE_NAME = 'ytdlp_plugins'
+_BASE_PACKAGE_PATH = Path(__file__).parent
+
+
+# Public APIs
+# Anything else is NOT public and no backwards compatibility is guaranteed
+__all__ = [
+    'directories',
+    'load_plugins',
+    'load_all_plugins',
+    'register_plugin_spec',
+    'add_plugin_dirs',
+    'set_plugin_dirs',
+    'disable_plugins',
+    'get_plugin_spec',
+    'PACKAGE_NAME',
+    'COMPAT_PACKAGE_NAME',
+]
+
+
+@dataclasses.dataclass
+class PluginSpec:
+    module_name: str
+    suffix: str
+    destination: Indirect
```

Comment:
Possible issue here:  Indirect is not public but the register plugin spec etc. plugins are

Reply:
Might need to define a general interface that is not necessary tied to Indirect

Reply:
Or we make the custom plugin spec stuff internal-only for now

Reply:
Internal only for now is fine. We can later work on public documentation and implementation efforts